lib/archethic/db/cassandra_impl.ex

```diff
@@ -78,48 +78,100 @@ defmodule ArchEthic.DB.CassandraImpl do
   @doc """
   Fetch the transaction chain by address and project the requested fields from the transactions
   """
-  @spec get_transaction_chain(binary(), list()) :: Enumerable.t()
-  def get_transaction_chain(address, fields \\ []) when is_binary(address) and is_list(fields) do
+  @spec get_transaction_chain(binary(), list()) :: {list(Transaction.t()), boolean(), binary()}
+  def get_transaction_chain(address, all_fields \\ [])
+      when is_binary(address) and is_list(all_fields) do
     start = System.monotonic_time()
+    # order matter first query options should be passed first
+    # then field options should be passed
+    {options, fields} = Enum.split_while(all_fields, &get_options_and_fields?(&1))
```

Comment:
I don't get this part, what is the link between options and fields.
i think it should be distinct.
Maybe pass an option parameter just for the query options